### PR TITLE
Node do maintenance

### PR DESCRIPTION
**Pull request by aleksei-burlakov**
The crmsh will change the cib.xml itself, instead of using the crm_attribute when changing the maintenance status of a node. Besides, it will inform the user if there are resources in the maintenance or is-manage mode.

### CODE REVIEW

**Comment by aleksei-burlakov** [comment]
@krig , @diegoakechi , @gao-yan please review this PR.

File: crmsh/ui_node.py [hunk 1]
```diff
@@ -9,9 +9,99 @@
...
+    if 'maintenance' == attr:
+        conflicting_attr = 'is-managed'
+    if 'is-managed' == attr:
+        conflicting_attr = 'maintenance'
+
+    # Override the resources on the node
+    resources = [x.node for x in cib_factory.cib_objects
+        if x.obj_type == "primitive"
+            or x.obj_type == "group"
+            or x.obj_type == "clone"]
```
Comment (liangxin1300) [inline crmsh/ui_node.py]: I would prefer this:
`resources = [x.node for x in cib_factory.cib_objects if x.obj_type in ["primitive", "group", "clone"]]`

File: crmsh/ui_node.py [hunk 2]
```diff
@@ -9,9 +9,99 @@
 from . import utils
 from . import xmlutil
 from .msg import common_err, syntax_err, no_prog_err, common_info, common_warn
```
Comment (liangxin1300) [inline crmsh/ui_node.py]: should be `common_error` here, or have errors like `NameError: name 'common_error' is not defined`

File: crmsh/ui_node.py [hunk 3]
```diff
@@ -9,9 +9,99 @@
...
+            xmlutil.set_attr(n, attr, value)
+    return True
+
+def set_node_attr(cluster_node_name, attr_name, value, commit=True):
+    """
+    Set an attribute for a node
+    """
+    node_obj = cib_factory.find_node(cluster_node_name)
+    if node_obj is None:
+        common_error("CIB is not valid!")
```
Comment (liangxin1300) [inline crmsh/ui_node.py]: should be `common_err` here

File: crmsh/ui_node.py [hunk 4]
```diff
@@ -9,9 +9,99 @@
...
+        return False
+
+    common_debug("set_node_attr: %s" % (node_obj.obj_id))
+
+    xml_node = node_obj.node
+    node_obj.set_updated()
+    if not update_xml_node(xml_node, attr_name, value):
+        common_error("Failed to update node attributes for %s" % (cluster_node_name))
```
Comment (liangxin1300) [inline crmsh/ui_node.py]: should be `common_err` here

File: crmsh/ui_node.py [hunk 5]
```diff
@@ -9,9 +9,99 @@
...
+        return False
+
+    if not commit:
+        return True
+
+    if not cib_factory.commit():
+        common_error("Failed to commit updates to %s" % (cluster_node_name))
```
Comment (liangxin1300) [inline crmsh/ui_node.py]: should be `common_err` here

File: crmsh/ui_node.py (hunk 1 above)
Comment (aleksei-burlakov) [reply crmsh/ui_node.py]: This will really be better.

File: crmsh/ui_node.py (hunk 4 above)
Comment (aleksei-burlakov) [reply crmsh/ui_node.py]: Done.

File: crmsh/ui_node.py (hunk 5 above)
Comment (aleksei-burlakov) [reply crmsh/ui_node.py]: Done.

**Comment by aleksei-burlakov** [comment]
@liangxin1300 thanks for the review, I have made the required changes.

File: crmsh/ui_node.py (hunk 1 above)
Comment (aleksei-burlakov) [reply crmsh/ui_node.py]: Done.

File: crmsh/ui_node.py (hunk 3 above)
Comment (aleksei-burlakov) [reply crmsh/ui_node.py]: Done.

**Pull request merged**